libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/ViewportDataSourceProcessor.kt

```diff
@@ -193,12 +193,34 @@ internal object ViewportDataSourceProcessor {
         return try {
             val currentStepFullPoints = currentStepProgress.stepPoints ?: emptyList()
             // todo bottom slice might not be needed since we always append the user location
-            TurfMisc.lineSliceAlong(
+            val lineSliceCoordinatesForLookaheadDistance = TurfMisc.lineSliceAlong(
                 LineString.fromLngLats(currentStepFullPoints),
                 distanceTraveledOnStepKM,
                 lookaheadDistanceForZoom,
                 TurfConstants.UNIT_KILOMETERS
             ).coordinates()
+            val outputCoordinates: MutableList<Point> = emptyList<Point>().toMutableList()
```

Comment:
nit: move this to a function 

``` kotlin
    private fun sliceCoordinatesForLookaheadDistance(
        currentStepProgress: RouteStepProgress,
        distanceTraveledOnStepKM: Double,
        lookaheadDistanceForZoom: Double
    ): List<Point> {
...
```

And these TurfExceptions should not be captured with try/catch. We should capture the issues with unit tests
``` kotlin
        return try {
            sliceCoordinatesForLookaheadDistance()
        } catch (e: TurfException) {
            LoggerProvider.logger.e(Tag(TAG), Message(e.message.toString()))
            emptyList()
        }
```

Reply:
100%